code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/InputBoxNotificationBuilder.kt

```diff
@@ -0,0 +1,302 @@
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Building an input box template push notification."
+        )
+        val packageName = context.packageName
+        val smallLayout = RemoteViews(packageName, R.layout.push_template_collapsed)
+        val expandedLayout = RemoteViews(packageName, R.layout.push_template_expanded)
+
+        // create a silent notification channel if needed
+        if (pushTemplate.isFromIntent == true && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationManager =
+                context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            AEPPushNotificationBuilder.setupSilentNotificationChannel(
+                notificationManager,
+                pushTemplate.getNotificationImportance()
+            )
+        }
+
+        // create the notification channel if needed
+        val channelIdToUse = AEPPushNotificationBuilder.createChannelIfRequired(
+            context,
+            pushTemplate.channelId,
+            pushTemplate.sound,
+            pushTemplate.getNotificationImportance()
+        )
+
+        // create the notification builder with the common settings applied
+        val notificationBuilder = AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            channelIdToUse,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            R.id.basic_expanded_layout
+        )
+
+        // get push payload data. if we are handling an intent then we know that we should be building a feedback received notification.
+        val imageUri =
+            if (pushTemplate.isFromIntent != true) pushTemplate.imageUrl else pushTemplate.feedbackImage
+        val pushImage = PushTemplateImageUtil.downloadImage(cacheService, imageUri)
+
+        if (pushImage != null) {
+            expandedLayout.setImageViewBitmap(R.id.expanded_template_image, pushImage)
+        } else {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "No image found for input box push template."
+            )
+            expandedLayout.setViewVisibility(R.id.expanded_template_image, View.GONE)
+        }
+
+        val bodyText =
+            if (pushTemplate.isFromIntent != true) pushTemplate.body else pushTemplate.feedbackText
+        smallLayout.setTextViewText(R.id.notification_title, pushTemplate.title)
+        smallLayout.setTextViewText(R.id.notification_body, pushTemplate.body)
+        expandedLayout.setTextViewText(R.id.notification_title, pushTemplate.title)
+        expandedLayout.setTextViewText(
+            R.id.notification_body_expanded, bodyText
+        )
+
+        // add an input box to capture user feedback
+        if (pushTemplate.isFromIntent != true && !pushTemplate.inputBoxReceiverName.isNullOrEmpty()) {
```

Comment:
nit: lets bail early here if push is not being built from intent
`if (pushTemplate.isFromIntent == true || pushTemplate.inputBoxReceiverName.isNullOrEmpty()) {
        return notificationBuilder
}`